qrcodecheckin.php

```diff
@@ -307,27 +311,46 @@ function qrcodecheckin_delete_image($code) {
  */
 function qrcodecheckin_civicrm_permission(&$permissions) {
   $prefix = ts('QR Code Checkin') . ': ';
-  $permissions[QRCODECHECKIN_PERM] = array(
+  $permissions[QRCODECHECKIN_PERM] = [
     $prefix . ts(QRCODECHECKIN_PERM),
     ts('Access the page presented by the QR Code and click to change participant status to attended'),
-  );
+  ];
 }
 
 /**
  * Implements hook_civicrm_tokens.
  */
 function qrcodecheckin_civicrm_tokens(&$tokens) {
-  $tokens['qrcodecheckin'] = array(
-    'qrcodecheckin.qrcode_url' => ts("URL to the QR code image file"),
-    'qrcodecheckin.qrcode_html' => ts("Block of HTML code with both QR code image and link"),
-  );
+  $qrcode_events = Civi::settings()->get('qrcode_events');
+  if (empty($qrcode_events)) {
+    return;
+  }
+  // There are QR enabled events so let's define tokens for each of them
+  $events = \Civi\Api4\Event::get(FALSE)
+    ->addSelect('id', 'title')
+    ->addWhere('is_active', '=', TRUE)
+    ->addWhere('start_date', '>', date('Y-m-d'))
```

Comment:
Often folks want to send QR codes in advance of the event. What if this was changed to say all events with an end date that is NULL or in the future?